### PR TITLE
HHH-13857 Avoid initialization when obtaining persistent class with Hibernate.getClass()

### PR DESCRIPTION
**Pull request by beikov**
https://hibernate.atlassian.net/browse/HHH-13857

### CODE REVIEW

**Comment by gavinking** [comment]
I mean, I'm not *against* it, but for 100% sure some users are going to get bitten by this.

**Comment by beikov** [comment]
We decided to add a new method, but I'm having a hard time coming up with a good name for it. Do you have a suggestion @gavinking?
I was thinking maybe `getLazyClass` for the lack of a better idea.

File: hibernate-core/src/main/java/org/hibernate/Hibernate.java [hunk 1]
```diff
@@ -259,6 +259,26 @@ public static <T> Class<? extends T> getClass(T proxy) {
 		return (Class<? extends T>) result;
 	}
 
+	/**
+	 * Get the true, underlying class of a proxied entity. This operation might
+	 * initialize a proxy by side effect.
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/Hibernate.java:264]: At the moment, the Javadoc between this method and `#getClass` are identical - how would users know the difference?

```suggestion
	 * Get the true, underlying class of a proxied entity. 
	 * <p/>
	 * Like {@link #getClass}, this operation might initialize a proxy by side effect.
	 * However, here the initialization is avoided if possible.  If the entity type is 
	 * defined with subclasses, the proxy will need to be initialized to properly
	 * determine the class.
```

**Comment by sebersole** [comment]
Looks good to me overall with the new method.  Not a fan of "lazy", although I do not have a better suggestion atm

**Pull request merged**